### PR TITLE
To build Trilinos_jll package

### PR DESCRIPTION
**Pull request by ven-k**
v"12.12.1"
Builds for 5 Linux-based  platforms:

- Linux(:i686, libc=:glibc),
- Linux(:x86_64, libc=:glibc),
- Linux(:aarch64, libc=:glibc),
- Linux(:armv7l, libc=:glibc, call_abi=:eabihf),
- Linux(:powerpc64le, libc=:glibc)

### CODE REVIEW

File: T/Trilinos/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Linux(:x86_64, libc=:glibc),
+    Linux(:aarch64, libc=:glibc),
+    Linux(:armv7l, libc=:glibc, call_abi=:eabihf),
+    Linux(:powerpc64le, libc=:glibc)
```
Comment (giordano) [inline T/Trilinos/build_tarballs.jl:35]:
```suggestion
    Platform("i686", "linux"; libc="glibc"),
    Platform("x86_64", "linux"; libc="glibc"),
    Platform("aarch64", "linux"; libc="glibc"),
    Platform("armv7l", "linux"; libc="glibc", call_abi="eabihf"),
    Platform("powerpc64le", "linux"; libc="glibc"),
```

**Comment by giordano** [comment]
This has lots of troubles:

* missing license
* files with clashing names due to case-sensitivy
* broken symlinks

**Comment by ViralBShah** [comment]
This is a really complex library to build...

**Comment by ViralBShah** [comment]
Oops closed mistakenly.

**Comment by ViralBShah** [comment]
For now let's just do x86_64 + glibc and disable all other platforms.

**Comment by ven-k** [comment]
Now that it is working for x86_64 + glibc should I go ahead for other Linux platforms or open this for review as it is? @ViralBShah 

Also, symlink for a `TrilinosReferenceGuide.html` is broken. It isn't a necessary file. So I have ignored that warning. @giordano

**Comment by ViralBShah** [comment]
Let's build on all platforms once and see how far it goes.

**Comment by ViralBShah** [comment]
Ok, it builds on everything except windows. @ven-k Can you update this PR and filter out windows from the platforms? After that we can merge it.

**Pull request merged**